File: app/src/main/java/com/datarockets/mnchkn/fragments/players/PlayerPresenterImpl.java
package com.datarockets.mnchkn.fragments.players;


public class PlayerPresenterImpl implements PlayerPresenter, PlayerInteractor.OnUpdatedListener {

    private PlayerView mPlayerView;
    private PlayerInteractor mPlayerinteractor;

    public PlayerPresenterImpl(PlayerView playerView, PlayerInteractor playerInteractor) {
        this.mPlayerView = playerView;
        this.mPlayerinteractor = playerInteractor;
    }

    @Override
    public void increaseLevelScore(int score) {
        mPlayerinteractor.increaseLevelScore(score, this);
    }

    @Override
    public void decreaseLevelScore(int score) {
        mPlayerinteractor.decreaseLevelScore(score, this);
    }

    @Override
    public void increaseStrengthScore(int score) {
        mPlayerinteractor.increaseStrengthScore(score, this);
    }

    @Override
    public void decreaseStrengthScore(int score) {
        mPlayerinteractor.decreaseStrengthScore(score, this);
    }

    @Override
    public void onUpdateLevel(int score) {
        mPlayerView.setLevelScore(String.valueOf(score));
    }

    @Override
    public void onUpdateStrength(int score) {
        mPlayerView.setStrengthScore(String.valueOf(score));
    }

}

File: app/src/main/kotlin/com/datarockets/mnchkn/fragments/players/PlayerPresenterImpl.kt
package com.datarockets.mnchkn.fragments.players


class PlayerPresenterImpl(private val mPlayerView: PlayerView, private val mPlayerinteractor: PlayerInteractor) : PlayerPresenter, PlayerInteractor.OnUpdatedListener {

    override fun increaseLevelScore(score: Int) {
        mPlayerinteractor.increaseLevelScore(score, this)
    }

    override fun decreaseLevelScore(score: Int) {
        mPlayerinteractor.decreaseLevelScore(score, this)
    }

    override fun increaseStrengthScore(score: Int) {
        mPlayerinteractor.increaseStrengthScore(score, this)
    }

    override fun decreaseStrengthScore(score: Int) {
        mPlayerinteractor.decreaseStrengthScore(score, this)
    }

    override fun onUpdateLevel(score: Int) {
        mPlayerView.setLevelScore(score.toString())
    }

    override fun onUpdateStrength(score: Int) {
        mPlayerView.setStrengthScore(score.toString())
    }

}
